pyqtgraph/parametertree/parameterTypes/slider.py

```diff
@@ -95,20 +95,21 @@ def optsChanged(self, param, opts):
             # This is fine, since errors are from the parent scope which will stabilize after the widget is
             # constructed anyway
             pass
-        span = opts.get('span', None)
-        if span is None:
-            step = opts.get('step', 1)
-            start, stop = opts.get('limits', param.opts['limits'])
-            # Add a bit to 'stop' since python slicing excludes the last value
-            span = np.arange(start, stop + step, step)
-        precision = opts.get('precision', 2)
-        if precision is not None:
-            span = span.round(precision)
-        self.span = span
-        self.charSpan = np.char.array(span)
-        w = self.slider
-        w.setMinimum(0)
-        w.setMaximum(len(span) - 1)
+        if not hasattr(self,'span') or 'limits' in opts or 'step' in opts or 'span' in opts:
```

Comment:
I feel it will be more readable/maintainable to move this check to after the `suffix` check, and early return on false